app/code/core/Mage/Catalog/Model/Product/Image.php

```diff
@@ -108,6 +109,11 @@ class Mage_Catalog_Model_Product_Image extends Mage_Core_Model_Abstract
     protected $_watermarkHeigth;
     protected $_watermarkImageOpacity = 70;
 
+    /**
+     * @var string directory
+     */
+    protected static $_baseMediaPath;
```

Comment:
this introduces an additional state to the Application which for some usecases needs a special cleanup.
see https://github.com/OpenMage/magento-lts/blob/d1c7b4bb8ad806b50e510c20db783861c6735f82/app/Mage.php#L258-L274

Usecases can be changes while doing unittests, or running the project inside an Application Server in a permanent process. Although this is are rather rare cases, so I just moved the target branch to 20.0.

Still we should think about a general solution, how to manage such a "cache" in a easy to cleanup way.